presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -58,23 +60,26 @@ public double getOutputRowCount()
      * Returns estimated data size.
      * Unknown value is represented by {@link Double#NaN}
      */
-    public double getOutputSizeInBytes(Collection<Symbol> outputSymbols)
+    public double getOutputSizeInBytes(Collection<Symbol> outputSymbols, Map<Symbol, Type> types)
     {
         requireNonNull(outputSymbols, "outputSymbols is null");
 
         return outputSymbols.stream()
-                .map(this::getSymbolStatistics)
-                .mapToDouble(this::getOutputSizeForSymbol)
+                .mapToDouble(symbol -> getOutputSizeForSymbol(getSymbolStatistics(symbol), types.get(symbol)))
                 .sum();
     }
 
-    private double getOutputSizeForSymbol(SymbolStatsEstimate symbolStatistics)
+    private double getOutputSizeForSymbol(SymbolStatsEstimate symbolStatistics, Type type)
     {
         double averageRowSize = symbolStatistics.getAverageRowSize();
         double numberOfNonNullRows = outputRowCount * (1.0 - firstNonNaN(symbolStatistics.getNullsFraction(), 0d));
         if (isNaN(averageRowSize)) {
-            // TODO take into consideration data type of column
-            return numberOfNonNullRows * DEFAULT_DATA_SIZE_PER_COLUMN;
+            if (type instanceof FixedWidthType) {
+                averageRowSize = ((FixedWidthType) type).getFixedSize();
+            }
+            else {
+                averageRowSize = DEFAULT_DATA_SIZE_PER_COLUMN;
```

Comment:
now the value `DEFAULT_DATA_SIZE_PER_COLUMN` should probably be changed to sth like 50, since fixed-size data types aren't part of this "average" any more